lib/dialects/postgres/data-types.js

```diff
@@ -52,24 +52,24 @@ DATE.prototype.toSql = function() {
 
 var INTEGER = function() {
   if (!(this instanceof INTEGER)) return new INTEGER();
-  BaseTypes.INTEGER.apply(this, arguments);
-
-  // POSTGRES does not support length on INTEGER
-  delete this._length;
-  delete this.options.length;
```

Comment:
It'll apply the arguments first and remove them afterwards again. PG does not support length, zerofill or unsigned and all possible arguments would have to be removed again.